paimon-core/src/main/java/org/apache/paimon/iceberg/manifest/IcebergConversions.java

```diff
@@ -74,8 +79,32 @@ public static ByteBuffer toByteBuffer(DataType type, Object value) {
             case DECIMAL:
                 Decimal decimal = (Decimal) value;
                 return ByteBuffer.wrap((decimal.toUnscaledBytes()));
+            case TIMESTAMP_WITHOUT_TIME_ZONE:
+                TimestampType timestampType = (TimestampType) type;
+                precision = timestampType.getPrecision();
+                timestamp = (Timestamp) value;
+                return convertTimestampWithPrecisionToBuffer(timestamp, precision);
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+                LocalZonedTimestampType localTimestampType = (LocalZonedTimestampType) type;
+                precision = localTimestampType.getPrecision();
+                timestamp = (Timestamp) value;
+                return convertTimestampWithPrecisionToBuffer(timestamp, precision);
+            case TIME_WITHOUT_TIME_ZONE:
+                Long time = ((Integer) value).longValue();
+                return ByteBuffer.allocate(8).order(ByteOrder.LITTLE_ENDIAN).putLong(0, time);
```

Comment:
Why do you cast it to an `Integer`?

From Paimon's `TimeType`:

Data type of a time WITHOUT time zone consisting of {@code hour:minute:second[.fractional]} with up to **nanosecond** precision.

From Iceberg's `time` type:

Stores **microseconds** from midnight in an 8-byte little-endian long.

Please read the docs very carefully. You cannot use this long value directly.

Reply:
Thank you very much. The test has passed. Please help me review the code again